content/source/docs/extend/debugging.html.md

```diff
@@ -0,0 +1,179 @@
+Terraform itself using the `TF_LOG` environment variable. Set it to `TRACE` to
+output all log lines, `DEBUG` to output all log lines except `TRACE` lines,
+etc.
+
+~> **NOTE:** prior to v0.15.0 of Terraform, levels besides `TRACE` may not be
+   entirely reliable. See [this
+   PR](https://github.com/hashicorp/terraform/pull/26632) for more details.
+
+By default, log output will be recorded to `stderr` when `TF_LOG` is set. To
+write log output to a file instead, set `TF_LOG_PATH` to a file's path. Log
+output will be appended there. It's worth noting that this file will not be
+truncated before log output is written.
+
+#### Separating Out Provider Logs
+
+Terraform can have some verbose logging and it can be difficult to parse log
+files when both Terraform and providers are writing to the same log file. To
+address this, following the 0.15.0 release of Terraform, the `TF_LOG_CORE`
+environment variable and `TF_LOG_PROVIDER` environment variable can now be set
+to different levels. Set either to `off` to turn off its logging entirely.
+
+## Debugger-Based Debugging
+
+Debugger-based debugging is a method of using debugging tools like
+[delve](https://github.com/go-delve/delve) to inspect what is happening in a
+provider as it is happening, often using breakpoints to pause execution of the
+provider and examine the values of variables. This method of debugging must be
+done contemporaneously; the developer doing the debugging needs to actively run
+Terraform using the appropriate configuration and in the appropriate
+environment to induce the behavior being examined. It is therefore most useful
+when a bug has been reliably reproduced. It does, however, enable developers to
+ask arbitrary questions and step through provider executions, exploring what is
+happening in the provider as part of the execution.
+
+~> **NOTE**: Debugger-based debugging only works with Terraform versions
+   0.12.26 and higher.
+
+### Enabling Debugging In A Provider
+
+Before a debugger can attach to a provider, the provider must be using version
+2.0.0 of the Terraform Plugin SDK. The plugin must also be started in debug
+mode, called `plugin.Debug` instead of `plugin.Serve`. It is recommended you
+enable this using a flag, as the provider should use `plugin.Serve` under
+normal operation, when not being debugged.
+
+```go
+func main() {
+	var debugMode bool
+
+	flag.BoolVar(&debugMode, "debug", false, "set to true to run the provider with support for debuggers like delve")
+	flag.Parse()
+
+	opts := &plugin.ServeOpts{ProviderFunc: provider.New}
+
+	if debugMode {
+		// TODO: update this string with the full name of your provider as used in your configs
+		err := plugin.Debug(context.Background(), "registry.terraform.io/my-org/my-provider", opts)
+		if err != nil {
+			log.Fatal(err.Error())
+		}
+		return
+	}
+
+	plugin.Serve(opts)
+}
+```
+
+It is important that a provider only be started in debug mode when you intend
```

Comment:
```suggestion
It is important to start a provider in debug mode only when you intend
```

Alternative to @nfagerlund's suggestion for those with a split-infinitive intolerance.